src/opencensus/proto/agent/metrics/v1/metrics_service.proto

```diff
@@ -42,6 +43,13 @@ message ExportMetricsServiceRequest {
 
   // A list of metrics that belong to the last received Node.
   repeated opencensus.proto.metrics.v1.Metric metrics = 2;
+
+  // The resource for the metrics in this message that do not have an explicit
+  // resource set.
+  // If unset, the most recently set resource in the RPC stream applies. It is
+  // valid to never be set within a stream, e.g. when no resource info is known
+  // at all or when all sent metrics have an explicit resource set.
+  opencensus.proto.resource.v1.Resource resource = 3;
```

Comment:
One intention for these protocol is to be used for example when collect data from multiple nodes on the same RPC stream:
{App1, App2} -> oc-agent1 -> oc-agent2

Let's say oc-agent1 receives:
* From App1 {Node1, Metrics1, Resource1}
* From App2 {Node2, Metrics2, null (no resource)}

Then oc-agent1 sends to oc-agent2:
* {Node1, Metrics1, Resource1}
* {Node2, Metrics2, null (no resource)}

Based on the current description Metrics2 will be Resource1 which is not correct.

Couple of suggestions:
1. If no-resource then we should send an empty Resource. or type = unknown.
2. Resources are cached per Node. So I would say "If unset, the most recently set resources in the RPC stream for the same Node ..."

Reply:
Agreeing with 1) – empty resource seems fine. I can't see any case where we'd want to distinguish an unset and an empty resource.

Suggestion 2. concerns me a bit, e.g. in particular this case:

* `1)` app1 -> {node1, metrics1, resource1}
* `2)` app2 -> {node2, metrics2, resource2}
* `3..999)` without app1 appearing
* `1000)` app1 -> {node1, metrics3, empty/unset}

It seems awkward to jump back to using `resource1` in step 1000. For long streams with lots of senders, this could become rather hard to reason about during debugging. It also requires the cache of a single connection to grow linearly with the number of apps multiplexed over it.

Would it make sense to slightly alter this to: "A change in `Node` always resets the most recent `Resource`"

Effectively, changing the node requires the resource to be changed too (or be considered empty after). This way, the node and resource cache of a connection have a constant size of 1 as well.